smithy-build/src/main/java/software/amazon/smithy/build/model/MavenRepository.java

```diff
@@ -26,10 +26,12 @@ public final class MavenRepository implements ToSmithyBuilder<MavenRepository> {
 
     private final String url;
     private final String httpCredentials;
+    private final Optional<String> id;
```

Comment:
so it'd be null by default? frankly I'd rather use Optional than allow a null and all the related problems.

Reply:
What related problems you see in this context? I don't like using optional as class attributes as they have some lacking features, such as not implementing `Serializable`, and would rather wrap the `null` in them only when consumers request the value, specially since allowing it to be `null` in this class context has no impact and will match the implementation of the `httpCredentials` attribute, which is also optional.

Reply:
In Java, the idiom is to store local properties on classes as null (and to accept nullable arguments in methods), and only wrap in Optional in getters. We use that idiom throughout the rest of the code base too.

Reply:
> What related problems you see in this context?

just NPEs. in any context ;)

Let's drop the Optional here, then.

Reply:
That's fair (: I agree it comes with a cost, but given we're wrapping it up with an Optional on the getter it seems low risk.